src/app/leverage/provider.tsx

```diff
@@ -239,6 +245,51 @@ export function LeverageProvider(props: { children: any }) {
     fetchStats()
   }, [baseToken])
 
+  useEffect(() => {
+    async function fetchCostOfCarry(
+      jsonRpcProvider: providers.JsonRpcProvider,
+    ) {
+      try {
```

Comment:
minor: would be nice to move all this logic in a separate file - as the provider is already so bloated.

Reply:
agreed - pulled into a separate file